Add redux to login

nicboy17 opened the pull request:
Connected login with redux

and refactored

nicboy17 committed:
server login fix

-switch username to email

nicboy17 committed:
add redux&saga to login

skassam21 commented on client/src/pages/Landing.js:
```diff
@@ -1,39 +1,38 @@
-import Profile from './Profile.js'
+import Login from './Login.js';
+import SignUp from './SignUp.js';
+import Profile from './Profile.js';
```
@mrchrischan @kevncheng @nicboy17 are you all using different linters for the front-end or did you add one more recently? If you are using visual code, you can have one that autocorrects (i.e. prettier.io) after you save

nicboy17 replied:
added to linter

skassam21 commented on client/src/pages/Login.js:
```diff
@@ -0,0 +1,82 @@
+                <div className={classes.modal}>
+                    <h1 className={classes.welcome}>Welcome!</h1>
+                    <div className={classes.emailContainer}>
+                        <TextField label='E-mail' className={classes.inputemail} onChange={handleChangeEmail}/>
```
can we make inputemail camelcase

nicboy17 replied:
fixed

skassam21 commented on client/src/pages/Login.js:
```diff
@@ -0,0 +1,82 @@
+                        <TextField label='E-mail' className={classes.inputemail} onChange={handleChangeEmail}/>
+                    </div>
+                    <div className={classes.passContainer}>
+                        <TextField label='Password' type="password" className={classes.inputpass} onChange={handleChangePass}/>
```
same with inputpass

nicboy17 replied:
fixed

skassam21 commented on client/src/pages/Login.js:
```diff
@@ -0,0 +1,82 @@
+    let [pass, postPass] = useState('');
+
+    const handleChangePass = (e) => {
+        postPass(pass = e.target.value);
```
why do we do: `postPass(pass = e.target.value)` vs. `postPass(e.target.value)`

nicboy17 replied:
not sure, i'll change that

skassam21 commented on client/src/pages/Login.js:
```diff
@@ -0,0 +1,82 @@
+    };
+
+    const postInfo = () => {
+        login({email, password:pass});
```
I feel like we need a space between pass and the colon (again if we used prettier.io, we could be consistent here)

nicboy17 replied:
fixed (added to linter)

skassam21 reviewed (commented):
comments on style of code

nicboy17 committed:
Merge branch 'master' into add-redux-to-login

# Conflicts:
#	client/package-lock.json
#	client/src/pages/Landing.js
#	client/src/pages/Login.js

nicboy17 committed:
refactoring

- added to linter
- moved images to assets
- removed landing/ping
- fixed login issues

nicboy17 committed:
make login default route

skassam21 commented on client/src/components/Dialog/PostDialog/PostDialog.js:
```diff
@@ -5,14 +5,14 @@ import DialogActions from '@material-ui/core/DialogActions';
 import DialogTitle from '@material-ui/core/DialogTitle';
 import { Chip } from '@material-ui/core';
-import CircularProgress from '@material-ui/core/CircularProgress';
+// import CircularProgress from '@material-ui/core/CircularProgress';
```
don't comment out code like this, delete it instead

ChrisTHChan replied:
yeah I noticed the validation stuff too. I'm working on integrating the boards and posts right now, but I can tackle the validation stuff later today

skassam21 commented on client/src/components/Dialog/PostDialog/PostDialog.js:
```diff
@@ -5,14 +5,14 @@ import DialogActions from '@material-ui/core/DialogActions';
 import FormContent from './FormContent';
 import FileUploader from './FileUploader';
-import BoardList from './BoardList';
+// import BoardList from './BoardList';
```
same

skassam21 commented on client/src/pages/Login.js:
```diff
@@ -1,187 +1,74 @@
-                        </Link>
-                    </p>
+                    <p className={classes.forgotPassword}>Forgot your password?</p>
+                    <button className={classes.loginButton} onClick={postInfo}>Login</button>
```
I noticed this in your project - is there any reason you are not using the material ui Button class or the Typography classes?

ChrisTHChan replied:
no reason, I'm just more familiar with classic css and html

skassam21 reviewed (commented):
Looks good overall - a lot of refactoring - @mrchrischan - you should take a look to since Nick refactored some of your code as well. I also made a comment on some older code that we didn't review before

However, I have noticed that this PR doesn't handle some validation issues:
- a user with same email/username signs up
- 400s for login 

Lastly, it doesn't do any routing when I finish the signup/login flow, is that intentional?
